feat: modernize dependencies and expand Python support

WilliamJudge94 opened the pull request:
This PR modernizes the project's dependencies and expands Python version support, addressing compatibility issues and updating deprecated libraries.

  ## Changes Made

  ### Python Version Support
  - **FEAT: adding python 3.13 to tests** - Added Python 3.13 to CI/CD test matrix for future compatibility

  ### Dependency Modernization
  - **FEAT: updating imghdr to use filetype instead** - Replaced deprecated `imghdr` with modern `filetype` library
  - **BUG: removing old imghdr tests** - Cleaned up obsolete test cases for removed dependency
  - **FEAT: using minify-html** - Replaced deprecated `htmlmin` with modern `minify-html` library

  ### HTML Processing Improvements
  - **BUG: use minify-html instead of htmlmin** - Migrated from `htmlmin` to more efficient `minify-html`
  - **BUG: kwargs difference from minify-html** - Fixed parameter compatibility issues with new HTML minifier
  - **BUG: change html search param for valid output** - Updated HTML parameter handling for correct output
  - **BUG: updating html checks due to minify-html** - Adjusted HTML validation logic for new minification library

  ## Testing Strategy

  - [x] Verify Python 3.13 compatibility across all test suites
  - [x] Confirm HTML processing works correctly with minify-html
  - [x] Validate filetype library functionality replaces imghdr completely

  ## Impact

  - **Compatibility**: Extends Python support to include 3.13
  - **Performance**: Improved HTML minification with more efficient library
  - **Maintenance**: Removes deprecated dependencies (imghdr, htmlmin)

  ## Breaking Changes

  None - all changes maintain backward compatibility while modernizing the underlying dependencies.

WilliamJudge94 commented:
@fabclmnt I apologize for not running the recommit beforehand. The commit messages and pre-commit checks should now be passing with the updates in 778d0aa554aafd0294664daf8ff30be43bf78d7c and 712732a8444dc40f9404b974d7695d960acf8386.

WilliamJudge94 commented:
@fabclmnt, based on the latest test results, there seems to be a couple of remaining issue :

1. `typing.io` does not exist anymore in Python 3.13. This affects the PySpark module. I can work towards updating the PySpark version for ydata-profiling. 

2. It appears as though a test data download link is broken. How would you like me to proceed? https://github.com/ydataai/ydata-profiling/actions/runs/17631150209/job/50623808947#step:9:3659

fabclmnt commented:
> @fabclmnt, based on the latest test results, there seems to be a couple of remaining issues:
> 
> 1. `typing.io` does not exist anymore in Python 3.13. This affects the PySpark module. I can work towards updating the PySpark version for ydata-profiling.
> 2. It appears as though a test data download link is broken. How would you like me to proceed? https://github.com/ydataai/ydata-profiling/actions/runs/17631150209/job/50623808947#step:9:3659

Hi @WilliamJudge94 ,

Thanks for your contribution! 

1. Yes, let’s go ahead and update the PySpark version and see if that resolves the issue. Since we need to support PySpark 4.0.0, some code adjustments may be required.

2. NASA has updated its dataset API. The current URL should be replaced with:
https://data.nasa.gov/docs/legacy/meteorite_landings/Meteorite_Landings.csv
instead of:
https://data.nasa.gov/api/views/gh4g-9sfh/rows.csv?accessType=DOWNLOAD

Really excited about this PR, thanks for pushing this forward!

WilliamJudge94 commented:
> > @fabclmnt, based on the latest test results, there seems to be a couple of remaining issues:
> > 
> > 1. `typing.io` does not exist anymore in Python 3.13. This affects the PySpark module. I can work towards updating the PySpark version for ydata-profiling.
> > 2. It appears as though a test data download link is broken. How would you like me to proceed? https://github.com/ydataai/ydata-profiling/actions/runs/17631150209/job/50623808947#step:9:3659
> 
> Hi @WilliamJudge94 ,
> 
> Thanks for your contribution!
> 
> 1. Yes, let’s go ahead and update the PySpark version and see if that resolves the issue. Since we need to support PySpark 4.0.0, some code adjustments may be required.
> 2. NASA has updated its dataset API. The current URL should be replaced with:
>    https://data.nasa.gov/docs/legacy/meteorite_landings/Meteorite_Landings.csv
>    instead of:
>    https://data.nasa.gov/api/views/gh4g-9sfh/rows.csv?accessType=DOWNLOAD
> 
> Really excited about this PR, thanks for pushing this forward!


1.  Updated PySpark to version ≥ 4.0. I couldn’t run the PySpark tests locally yet, but I’ll debug and add support for running them inside the .devcontainer.
2. Updated all URLs.
3. All tests pass locally, except for the PySpark tests that haven’t been run.


<img width="865" height="36" alt="image" src="https://github.com/user-attachments/assets/0c6cb9ad-f70a-461c-b8df-7f79094d9e1f" />

fabclmnt reviewed (approved):
🚀

emcek commented:
@WilliamJudge94 Some tests are failing (one timeout to nasa.gov and `ImportError` of 'typing')

fabclmnt commented:
> @WilliamJudge94 Some tests are failing (one timeout to nasa.gov and `ImportError` of 'typing')

Updated develop branch with the required fixes. We will be moving with the merge of this PR. 

@WilliamJudge94 Thank you for your contribution 🚀

WilliamJudge94 commented:
@fabclmnt You are very welcome! Thank you so much for the merge approval. Looking forward to the next release!
